tangelo/linq/tests/test_symbolic_simulator.py

```diff
@@ -0,0 +1,65 @@
+# Copyright 2023 Good Chemistry Company.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""A test class to check that the simulator class functionalities are behaving
+as expected for the symbolic backend.
+"""
+
+import unittest
+
+from tangelo.linq import Gate, Circuit
+from tangelo.helpers.utils import installed_backends
+
+from tangelo.helpers.utils import installed_backends
+from tangelo.linq.target.target_sympy import SympySimulator
+
+
+class TestSymbolicSimulate(unittest.TestCase):
+
+    @unittest.skipIf("sympy" not in installed_backends, "Test Skipped: Backend not available \n")
+    def test_simple_simulate(self):
+        """Test simulate of a simple rotation gate with a symbolic parameter."""
+
+        from sympy import symbols, cos, sin
+
+        simple_circuit = Circuit([Gate("RY", 0, parameter="alpha")])
+        backend = SympySimulator()
+        probs, _ = backend.simulate(simple_circuit, return_statevector=False)
+
+        alpha = symbols("alpha", real=True)
+
+        self.assertDictEqual(probs, {"0": (cos(alpha/2))**2, "1": (sin(alpha/2))**2})
```

Comment:
would it make sense to have a test that gets us the symbolic representation of a state resulting from a succession of a bunch of basic gates in our package (a circuit with 2 qubits), and verified the expression was correct by substituting the parameter(s) in the expression by real values and comparing to reference values ?

That would basically allow us to both test how this backend is doing with a non-trivial circuit trying a bunch of our gates, but also illustrate how to perform substitution in the resulting expression.

Reply:
yeah it would make sense, I can add one test.